Fix flaky rill yaml e2e

lovincyrus opened the pull request:
INSERT DESCRIPTION HERE

**Checklist:**
- [ ] Covered by tests
- [ ] Ran it and it works as intended
- [ ] Reviewed the diff before requesting a review
- [ ] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

ericpgreen2 commented:
I see the test is still flaky: 
<img width="1152" height="566" alt="image" src="https://github.com/user-attachments/assets/19436469-ef80-4fae-8476-1f6976c5751f" />

Do you know what the root cause is?

lovincyrus commented:
> I see the test is still flaky: <img alt="image" width="1152" height="566" src="https://private-user-images.githubusercontent.com/14206386/497967103-19436469-ef80-4fae-8476-1f6976c5751f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk3Nzg0MzcsIm5iZiI6MTc1OTc3ODEzNywicGF0aCI6Ii8xNDIwNjM4Ni80OTc5NjcxMDMtMTk0MzY0NjktZWY4MC00ZmFlLTg0NzYtMWY2OTc2YzU3NTFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA2VDE5MTUzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4YzBmYjM3NGZjNjEyYmQ5MjJmZjRmNTBiYmJiZTNkMTQ2ZDdmM2ZkNTI4NjBkM2Q1NjJmOWUzZDM4ZmY5NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4bb1uKpNGg6R1Egc8h_7i0lxYuJ98zW96gQgVv-KmdE">
> 
> Do you know what the root cause is?

There is a chance that it is because of 

```
2025-10-06T13:03:20.202	WARN	Reconcile failed	{"name": "clickhouse", "type": "Connector", "elapsed": "1.054s", "error": "clickhouse is not ready"}
```

I also see this locally

<img width="1742" height="852" alt="CleanShot 2025-10-06 at 13 09 53@2x" src="https://github.com/user-attachments/assets/ad47d516-ad32-4a6c-a54e-ccc68b4ff316" />

lovincyrus commented:
Where did you see the test failing? It's all green for me in this PR. @ericpgreen2

lovincyrus commented:
Removed the retries and added `waitForFileNavEntry` to wait for the file to exist properly 🚀 

<img width="3452" height="1860" alt="CleanShot 2025-10-07 at 16 58 40@2x" src="https://github.com/user-attachments/assets/d00c8e98-ee66-4b21-8bbc-47d5cf4aa706" />
